File: app/src/main/java/de/westnordost/streetcomplete/data/download/WifiAutoDownloadStrategy.java
package de.westnordost.streetcomplete.data.download;


import javax.inject.Inject;

import de.westnordost.streetcomplete.data.osm.persist.OsmQuestDao;
import de.westnordost.streetcomplete.data.tiles.DownloadedTilesDao;
import de.westnordost.streetcomplete.data.visiblequests.OrderedVisibleQuestTypesProvider;

public class WifiAutoDownloadStrategy extends AActiveRadiusStrategy
{
	@Inject public WifiAutoDownloadStrategy(
			OsmQuestDao osmQuestDB, DownloadedTilesDao downloadedTilesDao,
			OrderedVisibleQuestTypesProvider questTypes)
	{
		super(osmQuestDB, downloadedTilesDao, questTypes);
	}

	/** Let's assume that if the user is on wifi, he is either at home, at work, in the hotel, at a
	 *  café,... in any case, somewhere that would act as a "base" from which he can go on an
	 *  excursion. Let's make sure he can, even if there is no or bad internet.
     */

	@Override public int getQuestTypeDownloadCount()
	{
		return 1000;
	}

	@Override protected int getMinQuestsInActiveRadiusPerKm2()
	{
		return 500;
	}

	@Override protected int[] getActiveRadii()
	{
		// checks if either in 600 or 300m radius, there are enough quests.
		return new int[]{600, 300};
	}

	@Override protected int getDownloadRadius()
	{
		return 1200;
	}
}

File: app/src/main/java/de/westnordost/streetcomplete/data/download/WifiAutoDownloadStrategy.kt
package de.westnordost.streetcomplete.data.download


import javax.inject.Inject

import de.westnordost.streetcomplete.data.download.tiles.DownloadedTilesDao
import de.westnordost.streetcomplete.data.quest.VisibleQuestsSource
import de.westnordost.streetcomplete.data.visiblequests.OrderedVisibleQuestTypesProvider

/** Download strategy if user is on wifi */
class WifiAutoDownloadStrategy @Inject constructor(
    visibleQuestsSource: VisibleQuestsSource,
    downloadedTilesDao: DownloadedTilesDao,
    questTypes: OrderedVisibleQuestTypesProvider
) : AActiveRadiusStrategy(visibleQuestsSource, downloadedTilesDao, questTypes) {

    /** Let's assume that if the user is on wifi, he is either at home, at work, in the hotel, at a
     * café,... in any case, somewhere that would act as a "base" from which he can go on an
     * excursion. Let's make sure he can, even if there is no or bad internet.
     *
     * Since download size is almost unlimited, we can be very generous here.
     * However, Overpass is as limited as always, so the number of quest types we download is
     * limited as before  */

    override val questTypeDownloadCount = 1000
    override val minQuestsInActiveRadiusPerKm2 = 1000

    // checks if either in 2600 or 2300m radius, there are enough quests.
    override val activeRadii = intArrayOf(2600, 2300)
    override val downloadRadius = 2500
}
